scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/clazz/ClassDefinitionExtractor.scala

```diff
@@ -47,8 +50,9 @@ class ClassDefinitionExtractor(settings: ClassExtractionSettings) extends LazyLo
   ): Map[String, List[MethodDefinition]] = {
     val membersPredicate = settings.visibleMembersPredicate(clazz)
     val methods          = extractPublicMethods(clazz, membersPredicate, staticMethodsAndFields)
+    val extensionMethods = extractExtensionMethods(membersPredicate, staticMethodsAndFields)
     val fields           = extractPublicFields(clazz, membersPredicate, staticMethodsAndFields).mapValuesNow(List(_))
-    filterHiddenParameterAndReturnType(methods ++ fields)
+    filterHiddenParameterAndReturnType(methods ++ fields ++ extensionMethods)
```

Comment:
I think that we shouldn't add these extensions methods into `ClassDefinition`. It is not a thing that changes for each class. It is a thing that is added to every object. We should build some wrapper for `ClassDefinition` that would inject these methods in place of usage of `methods` method. Thanks to that, we will be flexible in further changes, e.g. we could add some extension methods only in some specific context of typing. Also, thanks to this approach we avoid hacks like `ClassDefinitionAssertions`

Reply:
Done